### PR TITLE
Upload PNGs

### PR DESCRIPTION
**Pull request by Reettaphant**
Allow for uploading pngs to the grid and send back data on the png type and the png size.

### CODE REVIEW

File: image-loader/app/lib/imaging/FileMetadataReader.scala [hunk 1]
```diff
@@ -78,17 +81,50 @@ object FileMetadataReader {
...
+          Dimensions(width, height)
+        }
+
+        case _ => None
+
       }
-      yield Dimensions(jpegDir.getImageWidth, jpegDir.getImageHeight)
     }
 
+  def exportPngDirectory(metadata: Metadata, mimeType: Option[String]): Map[String, String] = {
```
Comment (kenoir) [inline image-loader/app/lib/imaging/FileMetadataReader.scala]: I don't think you should have to hand the mimeType in here. Can you move the case statement up to the calling method?

File: image-loader/app/lib/imaging/FileMetadataReader.scala [hunk 2]
```diff
@@ -21,7 +22,7 @@ object FileMetadataReader {
   private implicit val ctx: ExecutionContext =
     ExecutionContext.fromExecutor(Executors.newCachedThreadPool)
 
-  def fromIPTCHeaders(image: File): Future[FileMetadata] =
+  def fromIPTCHeaders(image: File, mimeType: Option[String]): Future[FileMetadata] =
```
Comment (kenoir) [inline image-loader/app/lib/imaging/FileMetadataReader.scala]: Is this data were trying to extract in the IPTC headers? If not this method is misnamed, or extracting the PNG data needs to go in another method.

File: common-lib/src/main/scala/com/gu/mediaservice/model/FileMetadata.scala [hunk 3]
```diff
@@ -11,7 +11,9 @@ case class FileMetadata(
   xmp: Map[String, String]       = Map(),
   icc: Map[String, String]       = Map(),
   getty: Map[String, String]     = Map(),
-  colourModel: Option[String]    = None
+  colourModel: Option[String]    = None,
+  pngData: Map[String, String]   = Map()
```
Comment (kenoir) [inline common-lib/src/main/scala/com/gu/mediaservice/model/FileMetadata.scala]: Does this have to be specific to PNG data? Could we split the data collected into more generically named chunks?

**Comment by kenoir** [comment]
Nice work. Let's show @paperboyo in TEST before we deploy.

**Comment by paperboyo** [comment]
Amazing! So the metadata engine we use deals with PNG properties! Brilliant! Some should be shared between file formats (e.g. True Colour without alpha can be both some PNGs & all RGB JPGs). It also looks like we can not only extract textual metadata from PNG-specific fields, but *also* from the embedded XMP! (https://github.com/drewnoakes/metadata-extractor/issues/62#issuecomment-74349426). Good times!

**Comment by kenoir** [comment]
:+1: We should confirm this behaves in test (and doesn't disrupt the jpeg workflow).

**Comment by kenoir** [comment]
@Reettaphant if you're happy this works in test, feel free to push it to prod.

File: kahuna/public/js/services/api/media-cropper.js [hunk 4]
```diff
@@ -38,9 +38,11 @@ cropperApi.factory('mediaCropper', ['mediaApi', '$q', function(mediaApi) {
     function canBeCropped(image) {
         // Images can only be cropped if there is a link to the crops
         // TODO: this should be an Action
+        console.log('image ', image);
```
Comment (kenoir) [inline kahuna/public/js/services/api/media-cropper.js]: Can we take these out before pushing to PROD?

**Pull request merged**